File: backend/src/conferences/entities/conference.entity.ts
import { ApiProperty } from '@nestjs/swagger';
import { Conference } from 'prisma/.prisma/client';

export class ConferenceEntity implements Conference {
  @ApiProperty()
  id!: number;
  @ApiProperty()
  imageUrl!: string;
  @ApiProperty()
  title!: string;
}

File: mobile/lib/src/features/conferences/interactor/entities/conference_entity.dart
sealed class ConferenceEntity {}

class ConferenceTitleEntity extends ConferenceEntity {
  final String title;

  ConferenceTitleEntity({
    required this.title,
  });
}

class ConferenceEntryEntity extends ConferenceEntity {
  final int id;
  final String title;
  final String imageUrl;

  ConferenceEntryEntity({
    required this.id,
    required this.title,
    required this.imageUrl,
  });
}
